internal/core/worker/runtimewatcher/runtime_watcher_worker.go

```diff
@@ -108,33 +112,29 @@ func (w *runtimeWatcherWorker) mitigateDisruptions() error {
 			zap.String("scheduler", w.scheduler.Name),
 			zap.Error(err),
 		)
-		return err
+		return nil
 	}
-	mitigateForRoomsAmount := 0
-	if totalRoomsAmount >= 2 {
-		mitigateForRoomsAmount, err = w.roomStorage.GetRoomCountByStatus(w.ctx, w.scheduler.Name, game_room.GameStatusOccupied)
+	mitigationQuota := 0
+	// We only apply disruptions if there is more than 1 room in the scheduler
+	if totalRoomsAmount >= MinRoomsToApplyDisruption {
+		mitigationQuota, err = w.roomStorage.GetRoomCountByStatus(w.ctx, w.scheduler.Name, game_room.GameStatusOccupied)
 		if err != nil {
 			w.logger.Error(
 				"failed to get occupied rooms for scheduler",
 				zap.String("scheduler", w.scheduler.Name),
 				zap.Error(err),
 			)
-			return err
+			return nil
```

Comment:
why do we silently fail?

Reply:
This is a recoverable error, it might've been due to some problem in Redis connection. In that case we don't want to close/exit the watcher, just retry on next ticker